### ISSUE TITLE
Yam 2.7 Systematic error when trying to send a previously saved outgoing message

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#295]
**Originally by _JDuch@fulladsl.be_ on 2012-03-21 07:16:52 +0100**

___
## Summary
When i do save an outgoing message, save it and then try ot send it later, i systematically get an error messag of this type:

"Couldn't read from file
'Datas:YAM_2.7/Outgoing/QFzM8QAMXIw=.001,Q'.

The file doesn't seem to exist.

(03/20/2012 7:02 NM)"

I did install Yam2.7 in a new directory, then copied .config .spamdata .addressbook .folders  to  this drawer.

## Steps to reproduce
 1. Make a new message
 2. Save it
 3. Edit it later
 4. Send it
## Expected results
 The message to be sent to its destination
## Actual results

The message is not sent
 I get an error message of the form
"Couldn't read from file
'Datas:YAM_2.7/Outgoing/QFzM8QAMXIw=.001,Q'.

The file doesn't seem to exist.

(03/20/2012 7:02 NM)"

The message description appears in the "outgoing" folder and is preceded with an icon showing a  white "x" on a red shield 
## Notes
I did install Yam2.7 in a new directory, then copied .config .spamdata .addressbook .folders  to  this drawer.

I have this problem right from the start, after installing YAM 2.7

To send saved messages i do have to copy and paste the text, the subject & destionation to a New message and send it immediately

**Comment by tboeckel** [comment]
**Originally on 2012-04-26 17:03:17 +0200**

___
Are you going to provide the information I asked for? Otherwise I will close it as "works for me".

**Comment by jens-maus** [comment]
**Originally by _JDuch@fulladsl.be_ on 2012-04-26 22:57:50 +0200**

___
Content of the mail i did sent you on 21 mar 2012 below 
In a second one i said:
===== 2d message
having done more tests i can confirm that the error message systematically
replaces the 3 last characters of the file name RQE with Q
=====
This is the message appearing when i try to send a mail first saved in ongoing

===== 1st message
O just made an other unsuccessdfull try giving me the message
"Couldn't read from file
'Datas:YAM_2.7/Outgoing/QF2CcAAAXqU=.001,Q'.

The file doesn't seem to exist.

(03/21/2012 7:54 VM)"

Beneath the content of ongoing drawer. Most files are  still there because of
an unsuccesfull 'sent'

I do observe that the error message gives an incomplete file name  (R & E are
suppressed from the 3 last characters

7.Datas:YAM_2.7> list outgoing
Directory "outgoing" op woensdag 21-maa-12
QF2CcAAAXqU=.001,RQE            383 ----rw-d Heden     07:54:35
QF2CCQALV3A=.001,RQE            382 ----rw-d Heden     07:52:54
QFzM8QAMXIw=.001,RQE            500 ----rw-d Gisteren  19:02:06
QFxG1QANtNI=.001,RQ             479 ----rw-d Gisteren  09:27:49
QFccEwAA0So=.001,RQE          14068 ----rw-d vrijdag   11:29:03
QFN-yQAG71k=.001,RQE           4456 ----rw-d 13-maa-12 17:52:34
.fconfig                        355 ----rw-d 22-feb-12 17:43:31
7 bestanden - 20K bytes - 31 blocks gebruikt

**Comment by jens-maus** [comment]
**Originally by _anonymous_ on 2012-04-26 22:58:27 +0200**

___
I did sent 2 answers to trac@yam.ch same day 03/21/2012
I'll forward them again.

**Comment by tboeckel** [comment]
**Originally on 2012-04-27 09:13:13 +0200**

___
Replying to JDuch@…:
> O just made an other unsuccessdfull try giving me the message
> "Couldn't read from file
> 'Datas:YAM_2.7/Outgoing/QF2CcAAAXqU=.001,Q'.
> 
> The file doesn't seem to exist.
> 
> (03/21/2012 7:54 VM)"

> QF2CcAAAXqU=.001,RQE            383 ----rw-d Heden     07:54:35

It is quite obvious why YAM cannot find the file it is looking for. But what really puzzles me is the name. The "Q" is perfectly alright, as it is a queued mail. The "E" character is "partly" alright and will be added if sending the mail fails for any reason (including YAM being unable to read the  file). But what is wrong is the "R" character which identifies already read mails. Mails in the outgoing folder cannot be marked as "read" and will always be "unread".

Can you please check the settings of your outgoing folder? The type must be "sent mail". Since the outgoing folder is a standard folder the type cannot be modified using YAM's GUI. If the type is anything else but "sent mail" then please check the .fconfig file in the outgoing directory. It must contain the line "Type = 2". If the type is different then please change this with a text editor and restart YAM.

**Comment by jens-maus** [comment]
**Originally by _JDuch@fulladsl.be_ on 2012-04-30 21:00:37 +0200**

___
>  Can you please check the settings of your outgoing folder? The type must be "sent mail".Can you please check the settings of your outgoing folder? The type must be "sent mail".


That is the case (ghosted) 


>  Then please check the .fconfig file in the outgoing directory. It must contain the line "Type = 2".

That .fconfig file contains the line "Type = 2"

**Comment by tboeckel** [comment]
**Originally on 2012-05-01 22:35:30 +0200**

___
Replying to JDuch@…:
> >  Can you please check the settings of your outgoing folder? The type must be "sent mail".Can you please check the settings of your outgoing folder? The type must be "sent mail".
> 
> 
> That is the case (ghosted) 
> 
> 
> >  Then please check the .fconfig file in the outgoing directory. It must contain the line "Type = 2".
> 
> That .fconfig file contains the line "Type = 2"

Well, then I am out of ideas for the moment. Your settings look ok, but I am not able to reproduce this issue.

Could you please create a FOLDER log using the debug version of YAM? Basics about YAM's debugging facilities can be found in the [FAQ](http://faq.yam.ch/content/12/54/en/basics-about-yam_s-debugging-facilities.html).

Maybe you even try the latest YAM 2.8-dev which got some extended debug output in this respect.

**Comment by jens-maus** [comment]
**Originally by _JDuch@fulladsl.be_ on 2012-05-02 11:27:07 +0200**

___
I created a folder log, but maybe with the normal YAM2.7 version
I do have problems launching Sashimi which often freezes  (eg sashimi > ram:debug, Sashimi ASKSAVE)
Run >NIL: Sashimi >"CON:0/20/640/120/Sashimi/AUTO/CLOSE/WAIT/INACTIVE" ON NOPROMPT
gives me an output that i can save to clipboard.

I get the result below when i do sent a previously saved message, after having selected it in the outgoing folder, & having hit the 'edit' button (but not canging anything to the message) 

till i get the error message and close it

00:W:    Timer.c:176:timer[4]: already stopped
00:W:       Timer.c:176:timer[3]: already stopped
00:W:  Logfile.c:85:potential invalid fopen call with return (0x00000000, 0x00000001)
00:W: Timer.c:697:timer event received but no timer was processed!!!
00:W:     YAM_MAf.c:594:skipping index loading due to LoadedMode 2 for folder 'Outgoing'
00:W:   Logfile.c:85:potential invalid fopen call with return (0x00000000, 0x00000001)
00:D:     DockyIcon.c:222:set custom Docky icon 0 5a125980
00:D:     DockyIcon.c:241:Docky icon changed
00:W:  Logfile.c:85:potential invalid fopen call with return (0x00000000, 0x00000001)
00:D:   DockyIcon.c:222:set custom Docky icon 1 5a125a70
00:W: Timer.c:697:timer event received but no timer was processed!!!


The output below starts with the hitting of the 'new' button, & is generated during creation of the message, save it in the ongoing folder,  call it back with th 'edit' button, sending it, till closing the error message. 

00:W:  YAM_UT.c:4165:MapRawKey returned != 1 (00000000)
00:W:  YAM_UT.c:4165:MapRawKey returned != 1 (00000000)
00:W:  YAM_UT.c:4165:MapRawKey returned != 1 (00000000)
00:W:  YAM_UT.c:4165:MapRawKey returned != 1 (00000000)
00:W:  YAM_UT.c:4165:MapRawKey returned != 1 (00000000)
00:W:  YAM_UT.c:4165:MapRawKey returned != 1 (00000000)
00:W:  Logfile.c:85:potential invalid fopen call with return (0x00000000, 0x00000001)
00:W:   YAM_MAf.c:594:skipping index loading due to LoadedMode 2 for folder 'Outgoing'
00:W:    Timer.c:176:timer[5]: already stopped
00:W:     Timer.c:176:timer[4]: already stopped
00:W:     Timer.c:176:timer[3]: already stopped
00:W:     Timer.c:176:timer[4]: already stopped
00:W:      Timer.c:176:timer[3]: already stopped
00:W:     Timer.c:176:timer[4]: already stopped
00:W:  Timer.c:176:timer[4]: already stopped
00:W:   Timer.c:176:timer[3]: already stopped
00:W:     Timer.c:176:timer[4]: already stopped
00:W:    Timer.c:176:timer[4]: already stopped
00:W:       Timer.c:176:timer[3]: already stopped
00:W:  Logfile.c:85:potential invalid fopen call with return (0x00000000, 0
x00000001)
00:W: Timer.c:697:timer event received but no timer was processed!!!
00:W:     Timer.c:176:timer[4]: already stopped
00:D:     DockyIcon.c:222:set custom Docky icon 0 5922f120
00:D:     DockyIcon.c:241:Docky icon changed
01:W:    smtp.c:1046:potential invalid fopen call with return (0x00000000, 
0x00000001)
00:W:  Logfile.c:85:potential invalid fopen call with return (0x00000000, 0x00000001)
00:D:   DockyIcon.c:222:set custom Docky icon 1 5922f030
00:D:   DockyIcon.c:241:Docky icon changed

**Comment by tboeckel** [comment]
**Originally on 2012-05-02 16:11:24 +0200**

___
Replying to JDuch@…:
> I created a folder log, but maybe with the normal YAM2.7 version
> I do have problems launching Sashimi which often freezes  (eg sashimi > ram:debug, Sashimi ASKSAVE)
> Run >NIL: Sashimi >"CON:0/20/640/120/Sashimi/AUTO/CLOSE/WAIT/INACTIVE" ON NOPROMPT
> gives me an output that i can save to clipboard.

Then you did not read the entire FAQ entry carefully enough. Although YAM's debugging messages can be captured using Sashimi it is also possible to either let YAM print all debugging messages to the shell you started YAM from (keyword STDOUT) or to let YAM write the debugging messages directly to a file (keyword FILE:filename).

> 00:W:  Logfile.c:85:potential invalid fopen call with return (0x00000000, 0x00000001)

Your log file settings seems to be messed up. You have logging enabled but you seem to have no valid path for the log files configured. However, this is not the root of this issue.

Unfortunately this debug log does not contain any helpful information. That is why I asked you to use the nightly build of YAM 2.8-dev as this will output some more debugging information. It is possible to install YAM 2.7 and YAM 2.8-dev in parallel in two directories, each with its own mail database.

And could you post your .folders file here please? I am getting the impression that there is something wrong with your pathes.

**Comment by jens-maus** [comment]
**Originally by _anonymous_ on 2012-05-03 07:20:10 +0200**

___
>Your log file settings seems to be messed up. You have logging enabled but you seem to have no valid path for the log files configured. However, this is not the root of this issue.

I did check my settings in settings/security: for section log files i read
log files 
---
Mode Normal
Path Amiga OS4/utilities/YAM  ( i did change it to sys:utilities/YAM but no change in behavior)
Log all events checked
---
I don't see a logfile recently changed. There is a YAM.log with logs starting from 14/01/2012 and no more lines after 27/02/2012 with lines referring to a John Doe

I didn't find YAM 2.8-dev nightly build yet

**Comment by tboeckel** [comment]
**Originally on 2012-05-03 07:26:03 +0200**

___
Replying to anonymous:

Please log in before answering. Otherwise your answer may end up as spam and we have to manually accept each and every answer.

> I don't see a logfile recently changed. There is a YAM.log with logs starting
> from 14/01/2012 and no more lines after 27/02/2012 with lines referring to a John
> Doe

Well, shouldn't that ring a bell for you? If logging worked at some time but does no longer recently (perhaps since you switched the machine?) then there must be something wrong, if you like it or not.

> I didn't find YAM 2.8-dev nightly build yet

On this very page (yam.ch) there is a download section with the nightly builds right at the top. To make it short: [here](http://yam.ch/downloads). Just pick the version which suits you best.

**Comment by jens-maus** [comment]
**Originally by _anonymous_ on 2012-05-03 09:39:37 +0200**

___
 I did download 2.8-dev/debug for OS4: 
installed i in a new drawer, copied over all the ".something" files
The problem with sending previously saved files in outgoing has gone.
I guess debug info will not point to the 2.7 problem.

About the logfile: 

the logging did not work because the path indicated in the security settings was taken over from previous install and did no longer exist. After correcting this the 'ticket' problem persisted.

I will try how well the 2.8 debug version behaves in other respects.

**Comment by tboeckel** [comment]
**Originally on 2012-05-03 09:47:55 +0200**

___
Replying to anonymous:

Again, you forgot to log in. Please do that before answering the next time.

> installed i in a new drawer, copied over all the ".something" files
> The problem with sending previously saved files in outgoing has gone.
> I guess debug info will not point to the 2.7 problem.

Most probably not. You might want to try the regular non-debug build of YAM 2.8-dev. If that version does not show this issue then it was fixed already since the release of YAM 2.7. The consequence for you should be logical then: either stick with 2.7 until 2.8 is eventually released and live with the known limitation, or completely switch over to 2.8-dev if you are brave enough. Although we do our best to keep the development version as stable as possible we cannot guarantee this at any time. It might happen that YAM 2.8-dev kills your kitten ;)

**Comment by jens-maus** [comment]
**Originally by _JDuch@fulladsl.be_ on 2012-05-03 15:03:08 +0200**

___
Sorry for not logging in

Contrary to what i wrote 2.8Dev gets the "ticket" problem too.

I have tried to pin down when it appears on a new install.
I installed 2.7 a couple of times and gradually added my .dot files.
The problem appears when i substitute the config file.

Changing this file only partially, the problem  appears as soon as i changed the [TCP/IP] settings (which is the minimum needed to make a connection.) I don't see anything peculiar there (has always worked)

The only thing i noticed was a difference in the pre-selection setting (0  in the archive's setting) while it was 1 in mine. I remember having been surprised seeing more info when moving from 2.6p1 to 2.7 (I did not change consciously any setting then, just copied .config from 2.6p1)

What i did notice is that just copying some of the .dot files without changing some paths, ( gallery, logfile, ..etc) may create problems & even crashes

I would like to produce YAM2.8' debug output now

 YAM.debug ?
USER/K,PASSWORD/K,MAILDIR/K,PREFSFILE/K,NOCHECK/S,HIDE/S,DEBUG/S,MAILTO/K,SUBJECT/K,LETTER/K,ATTACH/M,NOIMGWARNING/S,NOCATALOG/S,NOSPLASHWINDOW/S

I don't see the STDOUT or FILE keyword here ?

**Comment by tboeckel** [comment]
**Originally on 2012-05-03 15:33:32 +0200**

___
Replying to JosDuchIt:

> The only thing i noticed was a difference in the pre-selection setting (0  in the archive's setting) while it was 1 in mine. I remember having been surprised seeing more info when moving from 2.6p1 to 2.7 (I did not change consciously any setting then, just copied .config from 2.6p1)

Preselection has absolutely nothing to do with this issue, because preselection will be applied during POP3 mail download, while your issue is about SMTP mail upload.
 
> What i did notice is that just copying some of the .dot files without changing some paths, ( gallery, logfile, ..etc) may create problems & even crashes

If something crashes then we are highly interested in details, but please open a new ticket, preferrably one ticket for each crash. And please be absolutely precise about which files you are talking and which file with which content exactly causes a crash. The log file cannot be the culprit for any crashes as it will be created or appended to. Any previous content is irrelevant.

>  YAM.debug ?
> USER/K,PASSWORD/K,MAILDIR/K,PREFSFILE/K,NOCHECK/S,HIDE/S,DEBUG/S,MAILTO/K,SUBJECT/K,LETTER/K,ATTACH/M,NOIMGWARNING/S,NOCATALOG/S,NOSPLASHWINDOW/S
> 
> I don't see the STDOUT or FILE keyword here ?  

Read the FAQ again, **PLEASE**! And read it completely, from top to bottom! Just must set the ENV variable "yamdebug" appropriately. I.e. "setenv yamdebug stdout,folder" will enable FOLDER debugging messages only and everything will be printed to the shell you started YAM from.

What you were referring to are the normal commandline options and tooltypes. These are available in the normal build as well.

**Comment by jens-maus** [comment]
**Originally on 2012-05-03 15:36:36 +0200**

___
Replying to JosDuchIt:

> I would like to produce YAM2.8' debug output now
> 
>  YAM.debug ?
> USER/K,PASSWORD/K,MAILDIR/K,PREFSFILE/K,NOCHECK/S,HIDE/S,DEBUG/S,MAILTO/K,SUBJECT/K,LETTER/K,ATTACH/M,NOIMGWARNING/S,NOCATALOG/S,NOSPLASHWINDOW/S
> 
> I don't see the STDOUT or FILE keyword here ?  

Sorry, but I think you still mix up things here. In fact, we have two different things here regarding debugging. one is the DEBUG command-line option you can use with the YAM executable. This however is supposed to output the TCP/IP traffic on stdout only. So it is meant for ordanary users to check why a certain TCP/IP related error occurred. This option is even available in stable releases.

The other thing is the possibility to set an environment variable "ENV:yamdebug" with certain content which will instruct the debugging version of YAM (the YAM.debug file included in every nightly build/development version) to output additional runtime information either on the serial output or on stdout or even into a file. That is what Thore is referring to and you should use a command like

```
setenv yamdebug "folder,stdout"
```

to output the debugging on stdout and include additional information on the "folder" module in YAM. and by using the following command

```
setenv yamdebug "folder,file:ram:t/yamdebug.txt"
```

it will even log into a file instead of outputting to stdout.

make sure to restart YAM.debug everything you change the content of the environment variable so that it will re-read the content of it. Also have a look at the [FAQ item](http://faq.yam.ch/content/12/54/en/basics-about-yam_s-debugging-facilities.html) Thore was referring because there we explain the runtime debugging of YAM in more detail.

**Comment by jens-maus** [comment]
**Originally by _anonymous_ on 2012-05-06 20:47:00 +0200**

___
I have tested using these debug possibilities, but i don't find something that looks helpfull.
However testing extensively with Snoopy i can confirm this:
- I don't observe (yet) any real difference between 2.7 & 2.8
- The problem appears as soon as i replace (copy over) in the  a new install's .config file the existing [TCP/IP] data with my own. The observation is valid for 2.7 & 2.8  (I will check now for 2.6p1)
- the error is not really systematic, even in a same YAM using session
For testing I now typically do prepare a number of messages & save them in outgoing.
First time i did this i observed that i could send nearly all of them, retrieving them one after the other from 'Outgoing', but the last (which then got the infamous "cross shield & is seen when listed as an RGE file.
Repeating this  i can say i have more succes in sending a saved message when there are more messages waiting in outgoing.

Now about the R, Q & E extension characters, it is clear that the R appears in the outgoing folder too.
as i checked with listing the folder. When you just save a new message to outgoing, it will appear in YAM window in bold (unread) and when checking with "list" has no R in the extension characters.
As soon as you click on its line in outoing folder, the line is unbolded & the R is added to its filename.
From the Snoopy trace after having hit 'sent'
- i do observe that in its Open() calls it is addressed successively as 
.. mode NEW extension RQ
.. mode OLD extension RQ
.. mode OLD extension Q

that last one just after a Rename probably to the  Q extension
Probably: because it is clipped by Snoopy and because

this last call is mentioned as o.k or FAIL (quite in correspondence of a succesfull or failed  sent)

When i spoke of crashes in a previous mail they were related to files pointed to in YAM prefs that were no longer present (pointing to a suppressed YAM install) If i remember well the .spamdata file

**Comment by tboeckel** [comment]
**Originally on 2012-05-06 21:08:11 +0200**

___
Well, now everything becomes clear. You have the embedded read pane active, while I prefer the separate read window. With the latter a mail in the outgoing folder cannot be "read", but only b e edited. With the embedded read pane a mail will always be displayed immediately, which is different from the separate read window.

However, I still am not able to reproduce the issue, even with the embedded read pane being active. No matter how fast I double click on a yet unread mail in the outgoing folder the mail always is sent successfully.

I will check in a modification which will not mark mails in the outgoing folder as read. Please check again with the next nightly build if this fixes the issue for you. If not the please reopen this ticket.

-*!!! IMPORTANT NOTE !!!**
Log in before adding any reply! PLEASE!!!! It cannot be that hard. Your constant anonymous replies are becomming really annoying.

> When i spoke of crashes in a previous mail they were related to files pointed to
> in YAM prefs that were no longer present (pointing to a suppressed YAM install)
> If i remember well the .spamdata file

Sorry, but I cannot believe that. YAM's spam filter does not care a bit about a missing or existing .spamdata file. If it exists, it will be used. If it is missing YAM will run with a completely untrained spam filter. This is just the same situation as when you switch on the spam filter for the very first time. If replacing or omitting a certain file results in a crash then please make sure that you **know** which file it was. Just "remembering well" is not enough, I'm affraid.

**Comment by tboeckel** [comment]
**Originally on 2012-05-06 21:11:21 +0200**

___
In (38620e8):
 * Timer.c, mui/ReadMailGroup.c: a just read mail will only be marked as such, if the current folder is not the outgoing folder. This modification solves the inconsistency between a separate read window and the embedded read pane. This closes #295.
